lib/fw.c

```diff
@@ -729,6 +705,40 @@ static int switchtec_fw_info_metadata_gen3(struct switchtec_dev *dev,
 	return 1;
 }
 
+static int switchtec_fw_part_info_gen3(struct switchtec_dev *dev,
+				       struct switchtec_fw_image_info *inf)
+{
+	int ret = 0;
+
+	switch (inf->part_id) {
+		case SWITCHTEC_FW_PART_ID_G3_BOOT:
+			inf->part_addr = SWITCHTEC_FLASH_BOOT_PART_START;
+			inf->part_len = SWITCHTEC_FLASH_PART_LEN;
+			inf->active = true;
+			break;
+		case SWITCHTEC_FW_PART_ID_G3_MAP0:
+			inf->part_addr = SWITCHTEC_FLASH_MAP0_PART_START;
+			inf->part_len = SWITCHTEC_FLASH_PART_LEN;
+			ret = switchtec_fw_map_get_active(dev, inf);
+			break;
+		case SWITCHTEC_FW_PART_ID_G3_MAP1:
+			inf->part_addr = SWITCHTEC_FLASH_MAP1_PART_START;
+			inf->part_len = SWITCHTEC_FLASH_PART_LEN;
+			ret = switchtec_fw_map_get_active(dev, inf);
+			break;
+		default:
+			ret = switchtec_flash_part(dev, inf, inf->part_id);
+	}
+
+	if (ret)
+		return ret;
+
+	if (inf->part_id == SWITCHTEC_FW_PART_ID_G3_NVLOG)
```

Comment:
Small thing but this check was duplicated: it's in switchtec_fw_info_metadata_gen3().

Reply:
Yes, should be removed.